tools/cache_perf_tool/main.go

```diff
@@ -167,13 +181,59 @@ outerloop:
 	workerFunc := func() (shouldRetry bool, err error, val interface{}) {
 		return numGoroutines.Load() != int32(0), nil, val
 	}
-	err, _ = base.RetryLoop(parentCtx, "wait for writing goroutines to stop", workerFunc, base.CreateSleeperFunc(200, 100))
+	err, _ = base.RetryLoop(parentCtx, "wait for writing goroutines to stop", workerFunc, base.CreateSleeperFunc(500, 100))
 	if err != nil {
 		log.Printf("Error waiting for stat value (%d) to reach 0: %v", numGoroutines.Load(), err)
 	}
 
 }
 
+func printJavaPropertiesFile(ctx context.Context, dbContext *db.DatabaseContext) {
+	// Print the Java properties file to stdout
+	dbContext.UpdateCalculatedStats(ctx)
+	dbStats := dbContext.DbStats
+	// calculate here avg time to cache seq in ms
+	count := dbStats.Database().DCPCachingCount.Value()
+	timeNano := dbStats.Database().DCPCachingTime.Value()
+	avgTimeNano := float64(timeNano) / float64(count)
+	avgTimeMs := avgTimeNano / 1e6
+	_, _ = fmt.Fprintf(os.Stdout, "high_seq_feed=%d\npending_seq_len=%d\nhigh_seq_stable=%d\ncurrent_skipped_seq_count=%d\nnum_skipped_seqs=%d\n"+
+		"skipped_seq_len=%d\nskipped_seq_cap=%d\ndcp_caching_count=%d\ndcp_caching_time=%d\navg_time_per_seq_ms=%f\n",
+		dbStats.Database().HighSeqFeed.Value(), dbStats.Cache().PendingSeqLen.Value(), dbStats.Cache().HighSeqStable.Value(), dbStats.Cache().NumCurrentSeqsSkipped.Value(),
+		dbStats.Cache().NumSkippedSeqs.Value(), dbStats.Cache().SkippedSeqLen.Value(), dbStats.Cache().SkippedSeqCap.Value(), count,
+		timeNano, avgTimeMs)
+}
+
+func csvStats(ctx context.Context, dbContext *db.DatabaseContext) {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	numGoroutines.Add(1)
+	defer numGoroutines.Add(-1)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			once.Do(func() {
+				_, _ = fmt.Fprintf(os.Stderr, "timestamp,high_seq_feed,pending_seq_len,high_seq_stable,current_skipped_seq_count,"+
+					"num_skipped_seqs,skipped_seq_len,skipped_seq_cap,dcp_caching_count,dcp_caching_time,avg_time_per_seq_ms\n")
+			})
+			dbContext.UpdateCalculatedStats(ctx)
+			dbStats := dbContext.DbStats
+			// calculate here avg time to cache seq in ms
+			count := dbStats.Database().DCPCachingCount.Value()
+			timeNano := dbStats.Database().DCPCachingTime.Value()
+			avgTimeNano := float64(timeNano) / float64(count)
+			avgTimeMs := avgTimeNano / 1e6
+			_, _ = fmt.Fprintf(os.Stderr, "%s,%d,%d,%d,%d,%d,%d,%d,%d,%d,%f\n",
+				time.Now().Format(time.RFC3339), dbStats.Database().HighSeqFeed.Value(), dbStats.Cache().PendingSeqLen.Value(), dbStats.Cache().HighSeqStable.Value(),
```

Comment:
It's slightly easier to plot these if the time is a numerical timestamp, since you don't need to fiddle around with date formatting in a spreadsheet to get it parsing correctly.

`time.Now().Unix()`